### ISSUE TITLE
The `spack external` command does not honor scopes

### ISSUE THREAD

**Issue by glennpj** [LLNL/spack#18423]
The `spack external find` command does not honor scope. The expectation is this would be at least comparable to `spack compiler find`, which takes a `--scope` argument.

### Steps to reproduce the issue

```console
$ spack external find --scope site
==> Error: unrecognized arguments: --scope
```
Using `--config-scope` does not work either, writing packages.yaml to my home directory.
```console
$ spack --config-scope etc/spack/ external find
==> The following specs have been detected on this system and added to /Users/gpjohnsn/.spack/packages.yaml
autoconf@2.69    bison@3.7.1       flex@2.5.37  gmake@4.3      openssl@1.0.2k-fips  python@2.7.5
automake@1.13.4  cmake@2.8.12.2    flex@2.6.4   go@1.13.6      openssl@1.1.1g       python@2.7.18
automake@1.16.2  cmake@3.18.1      gcc@4.8.5    libtool@2.4.2  perl@5.16.3          python@3.7.9
bash@4.2.46      cuda@11.0.194     git@1.8.3.1  libtool@2.4.6  perl@5.30.3          python@3.8.5
bash@5.0.18      findutils@4.5.11  git@2.28.0   m4@1.4.16      pkg-config@0.27.1    texinfo@5.1
bison@3.0.4      findutils@4.7.0   gmake@3.82   m4@1.4.18      pkgconf@1.7.3        texinfo@6.7
```

### Information on your system

```console
$ spack debug report
* **Spack:** 0.11.2-8457-586fbe05b
* **Python:** 3.7.9
* **Platform:** linux-centos7-skylake_avx512
```

### Additional information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

**Comment by alalazo** [comment]
I think this was solved by @becker33 in #18165

**Comment by glennpj** [comment]
Yes, it was.
